simulator/src/main/java/com/hedera/block/simulator/generator/BlockAsFileLargeDataSets.java

```diff
@@ -0,0 +1,95 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.block.simulator.generator;
+
+import static com.hedera.block.simulator.generator.Utils.readFileBytes;
+import static java.lang.System.Logger.Level.INFO;
+
+import com.hedera.block.simulator.config.data.BlockStreamConfig;
+import com.hedera.block.simulator.config.types.GenerationMode;
+import com.hedera.hapi.block.stream.Block;
+import com.hedera.hapi.block.stream.BlockItem;
+import com.hedera.pbj.runtime.ParseException;
+import com.hedera.pbj.runtime.io.buffer.Bytes;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.io.File;
+import java.io.IOException;
+import javax.inject.Inject;
+
+/** A block stream manager that reads blocks from files in a directory. */
+public class BlockAsFileLargeDataSets implements BlockStreamManager {
+
+    private final System.Logger LOGGER = System.getLogger(getClass().getName());
+
+    private final String blockstreamPath;
+    private int currentBlockIndex = 0;
+    private int currentBlockItemIndex = 0;
+
+    private Block currentBlock = null;
+    private final String formatString;
+
+    /**
+     * Constructs a new BlockAsFileLargeDataSets instance.
+     *
+     * @param config the block stream configuration
+     */
+    @Inject
+    public BlockAsFileLargeDataSets(@NonNull BlockStreamConfig config) {
+        this.blockstreamPath = config.folderRootPath();
+        this.formatString = "%0" + config.paddedLength() + "d" + config.fileExtension();
+    }
+
+    @Override
+    public GenerationMode getGenerationMode() {
+        return GenerationMode.DIR;
+    }
+
+    @Override
+    public BlockItem getNextBlockItem() throws IOException, ParseException {
+        if (currentBlock != null && currentBlock.items().size() > currentBlockItemIndex) {
+            return currentBlock.items().get(currentBlockItemIndex++);
+        } else {
+            currentBlock = getNextBlock();
+            if (currentBlock != null) {
+                currentBlockItemIndex = 0; // Reset for new block
+                return getNextBlockItem();
+            }
+        }
+
+        return null; // No more blocks/items
+    }
+
+    @Override
+    public Block getNextBlock() throws IOException, ParseException {
+        currentBlockIndex++;
+
+        String nextBlockFileName = String.format(formatString, currentBlockIndex);
+        File blockFile = new File(blockstreamPath, nextBlockFileName);
+
+        if (blockFile.exists()) {
+            byte[] blockBytes = readFileBytes(blockFile.toPath());
+
+            LOGGER.log(INFO, "Loading block: " + blockFile.getName());
+
+            Block block = Block.PROTOBUF.parse(Bytes.wrap(blockBytes));
+            LOGGER.log(INFO, "block loaded with items size= " + block.items().size());
+            return block;
+        }
```

Comment:
we already know that the file exists, since we first calculate the next filename string, and based on that we check if the file exists. so really no need to handle this possibility here.